toolkit/scripts/setuplkgtoolchain.sh

```diff
@@ -23,39 +28,97 @@ get_lkg() {
 }
 
 check_for_modified_manifests() {
+    auto=$1
+    force=$2
     if git status --porcelain | grep -qP "(pkggen_core|toolchain)_$(uname -p).txt"; then
-        echo "Local modifications to 'pkggen_core_$(uname -p).txt' or 'toolchain_$(uname -p).txt' detected."
-        echo -e "\nNOTE: Changes to manifests were detected, and these will be overwritten. Hit Ctrl+C within 10 seconds to cancel...\n"
-        sleep 10s
+        if $force; then
+            echo "Forcing overwrite of local changes to 'pkggen_core_$(uname -p).txt' or 'toolchain_$(uname -p).txt'" 1>&2
+        else
+            if $auto; then
+                echo "Local modifications to 'pkggen_core_$(uname -p).txt' or 'toolchain_$(uname -p).txt' detected, will not auto update without --force" 1>&2
+                exit 1
+            else
+                echo "Local modifications to 'pkggen_core_$(uname -p).txt' or 'toolchain_$(uname -p).txt' detected." 1>&2
+                echo -e "\nNOTE: Changes to manifests were detected, and these will be overwritten. Hit Ctrl+C within 10 seconds to cancel...\n" 1>&2
+                sleep 10s
+            fi
+        fi
     fi
 }
 
-update_manifests() {
-    wget -nv https://raw.githubusercontent.com/microsoft/CBL-Mariner/$GIT_COMMIT/toolkit/resources/manifests/package/toolchain_$(uname -p).txt -O $ROOT_FOLDER/toolkit/resources/manifests/package/toolchain_$(uname -p).txt
-    wget -nv https://raw.githubusercontent.com/microsoft/CBL-Mariner/$GIT_COMMIT/toolkit/resources/manifests/package/pkggen_core_$(uname -p).txt -O $ROOT_FOLDER/toolkit/resources/manifests/package/pkggen_core_$(uname -p).txt
+# We don't want to update the timestamps unless we actually change the file, so download to a temp file and then move it over
+# if the files are different
+# $1 - filename (e.g. toolchain_aarch64.txt)
+update_manifest() {
+    local filename
+    local url
+    local filepath
+    filename="$1"
+    url="https://raw.githubusercontent.com/microsoft/CBL-Mariner/$GIT_COMMIT/toolkit/resources/manifests/package/${filename}"
+    filepath="$ROOT_FOLDER/toolkit/resources/manifests/package/${filename}"
+    filepath_tmp="${filepath}.tmp"
+
+    echo "Downloading $filename manifest from $url" 1>&2
```

Comment:
OK, now I'm really curious why we want to print everything to `stderr`. I'm not saying we don't, I just want to understand.:)

Reply:
When doing `variable = $(shell some_command.sh arg)`, all output on `stdout` will be sent to the variable. That means that any helpful debug output or status to the user needs to be on `stderr`. We can check the return with `ifneq ($(.SHELLSTATUS),0)...` but that won't explain why it failed,  just that it was non-zero.